### ISSUE TITLE
End 2 end testing

### ISSUE THREAD

**Issue by thomas-tran** [aviarytech/wacinsteins-monster#141]
Hi @brianorwhatever , do you have an end to end testing example for this?

**Comment by brianorwhatever** [comment]
hi @thomas-tran we have some cypress tests that run through the full 4 step waci-presentation-exchange flow. 

You can find the test file here https://github.com/aviarytech/wacinsteins-monster/blob/main/tests/cypress/integration/waci-pex.spec.js and there is a video demo of it running here https://github.com/aviarytech/wacinsteins-monster/tree/main/frontend/public/assets/videos

I have also allowed anyone to sign in using their email to the demo located here https://aviary.one/ and here https://aviary.vc/ which lets you play around with sending presentations back and forth between domains.

Let me know if you have any troubles or more questions.

**Comment by thomas-tran** [comment]
Thanks. I will take a look

**Comment by thomas-tran** [comment]
@brianorwhatever I've checked the code and just want to know how does the holder using the mobile wallet to derived the credential using the rest API presentations? Does the Rest API kms hold the private key of the holder?

**Comment by brianorwhatever** [comment]
yes these are only example credentials and the server acts as the holder. this is just for demonstration purposes

**Comment by Vanderscycle** [comment]
@thomas-tran this repo, from an outsider perspective, looks as if it fell in a state of disrepair, but we haven't forgotten about it as we are building on top of it on another project.

**Comment by thomas-tran** [comment]
@brianorwhatever One more question, do you have sample how to issue the credential to the wallet holder using the didcomm? I did not see it in any test cases.

**Comment by brianorwhatever** [comment]
hi @thomas-tran this project was created to implement the waci-pex spec (https://identity.foundation/waci-presentation-exchange/) which at the time did not include issuance so we only use example credentials here. there is no issuance
